src/virtual-machine.js

```diff
@@ -329,9 +329,8 @@ class VirtualMachine extends EventEmitter {
             }
             if (newName && RESERVED_NAMES.indexOf(newName) === -1) {
                 const names = this.runtime.targets
-                    .filter(runtimeTarget => runtimeTarget.isSprite())
+                    .filter(runtimeTarget => runtimeTarget.isSprite() && runtimeTarget.id !== target.id)
                     .map(runtimeTarget => runtimeTarget.sprite.name);
-
                 sprite.name = StringUtil.unusedName(newName, names);
```

Comment:
Another weird thing about thing about this bug was

"Every number but 2 is changed to 2, except for 2 itself, which becomes 3 when a user enters a string ending in a single 2 as a new sprite name."

Not needed for this fix, but is it weird that when you try to name two sprites TurboBlaster9000, the second one gets renamed to TurboBlaster2? Maybe it would make more sense to go to TurboBlaster9001 or TurboBlaster9000_2?